### PR TITLE
Port the workflows library

### PR DESCRIPTION
**Pull request by vladsavelyev**
Merging in the `cpg_pipes` library (under `cpg_utils.workflows`).

### CODE REVIEW

File: cpg_utils/flows/slack.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+"""
+Sending notifications to Slack. To enable, create a channel, add "Seqr Loader" app
+into a channel with:
+
+/invite @Seqr Loader
```
Comment (lgruen) [inline cpg_utils/flows/slack.py]: This is probably more generic now? Is there already a generic bot?

File: cpg_utils/flows/metamist.py [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+    id: str
+    sample_id: str
+    meta: dict
+    sequencing_type: str
+    alignment_input: AlignmentInput | None = None
+
+    @staticmethod
+    def parse(data: dict, check_existence: bool = False) -> 'MmSequence':
+        """
+        Parse dictionary to create a SmSequence object.
```
Comment (lgruen) [inline cpg_utils/flows/metamist.py]: The mix between `Mm` and `Sm` is a little confusing, but there's probably no great way around this until everything has been renamed on the Metamist side.

File: cpg_utils/flows/status.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        jobs: list[Job] | None = None,
+        prev_jobs: list[Job] | None = None,
+        meta: dict | None = None,
+    ) -> list[Job]:
+        """
+        Add Hail Batch jobs that update the analysis status.
+        """
+
+    @abstractmethod
+    def create_analysis(
```
Comment (lgruen) [inline cpg_utils/flows/status.py]: @illusional I wonder if you had any thoughts on this, based on our chat yesterday.

**Comment by lgruen** [comment]
> Merging in the `cpg_pipes` library (under `cpg_utils.flows`).

The actual classes are called `Workflow` -- would `cpg_utils.workflows` be more consistent or is that too long for your taste?

**Review by lgruen — APPROVED** [review]

File: cpg_utils/flows/metamist.py (hunk 2 above)
Comment (vladsavelyev) [reply cpg_utils/flows/metamist.py]: Renamed to simply `Sequence`

File: cpg_utils/flows/slack.py (hunk 1 above)
Comment (vladsavelyev) [reply cpg_utils/flows/slack.py]: Good question, I should look into creating a generic bot

File: cpg_utils/flows/status.py (hunk 3 above)
Comment (illusional) [reply cpg_utils/flows/status.py]: Not fully formed yet, I'll try to summarise all my thoughts into a doc soon based on chatting to everyone.

**Pull request merged**